src/find-best-model/src/main/scala/EvaluationUtils.scala

```diff
@@ -0,0 +1,128 @@
+import org.apache.spark.ml.param.{Param, ParamMap}
+import org.apache.spark.ml.regression._
+
+trait HasEvaluationMetric extends Wrappable {
+  /** Metric to evaluate the models with. Default is "all"
+    *
+    * The metrics that can be chosen are:
+    *
+    *   For Binary Classifiers:
+    *     - AreaUnderROC
+    *     - AUC
+    *     - accuracy
+    *     - precision
+    *     - recall
+    *
+    *   For Regression Classifiers:
+    *     - mse
+    *     - rmse
+    *     - r2
+    *     - mae
+    *
+    *   Or, for either type of classifier:
+    *     - all - This will report all the relevant metrics
+    *
+    * @group param
+    */
+  val evaluationMetric: Param[String] = StringParam(this, "evaluationMetric", "Metric to evaluate models with",
+    (s: String) => Seq(ComputeModelStatistics.MseSparkMetric,
+      ComputeModelStatistics.RmseSparkMetric,
+      ComputeModelStatistics.R2SparkMetric,
+      ComputeModelStatistics.MaeSparkMetric,
+      ComputeModelStatistics.AccuracySparkMetric,
+      ComputeModelStatistics.PrecisionSparkMetric,
+      ComputeModelStatistics.RecallSparkMetric,
+      ComputeModelStatistics.AucSparkMetric) contains s)
+  /** @group getParam */
+  def getEvaluationMetric: String = $(evaluationMetric)
+  /** @group setParam */
+  def setEvaluationMetric(value: String): this.type = set(evaluationMetric, value)
+  // Set default evaluation metric to accuracy
+  setDefault(evaluationMetric -> ComputeModelStatistics.AccuracySparkMetric)
+}
+
+object EvaluationUtils {
+  val modelTypeUnsupportedErr = "Model type not supported for evaluation"
+  // Find type of trained models
+  def getTransformerModelType(model: Transformer):String = {
+    model match {
+      case _: TrainedRegressorModel => SchemaConstants.RegressionKind
+      case _: TrainedClassifierModel => SchemaConstants.ClassificationKind
+      case evm: BestModel => getTransformerModelType(evm.getBestModel)
+      case _: ClassificationModel[_, _] => SchemaConstants.ClassificationKind
+      case _: RegressionModel[_, _] => SchemaConstants.RegressionKind
+      case _ => throw new Exception(modelTypeUnsupportedErr)
+    }
+  }
+
+  def getEstimatorModelType(model: Estimator[_]): String = {
+    model match {
+      case _: TrainRegressor => SchemaConstants.RegressionKind
+      case _: TrainClassifier => SchemaConstants.ClassificationKind
+      case _: Classifier[_, _, _] => SchemaConstants.ClassificationKind
+      case _: LinearRegression => SchemaConstants.RegressionKind
+      case _: DecisionTreeRegressor => SchemaConstants.RegressionKind
+      case _: GBTRegressor => SchemaConstants.RegressionKind
+      case _: RandomForestRegressor => SchemaConstants.RegressionKind
+      case _ => throw new Exception(modelTypeUnsupportedErr)
+    }
+  }
+
+  def getMetricWithOperator(model: Transformer, evaluationMetric: String): (String, Ordering[Double]) = {
+    val modelType = getTransformerModelType(model)
```

Comment:
Both of these should be in one function that takes in pipeline stages and case matches internally, likewise for getXModelType

Reply:
done!